src/app/components/filebrowseruss/filebrowseruss.component.ts

```diff
@@ -1014,6 +1014,10 @@ export class FileBrowserUSSComponent implements OnInit, OnDestroy {//IFileBrowse
       ); 
   }
 
+  refreshFileMetadatdaUsingPath(path: string) {
+    this.refreshFileMetadata(this.findNodeByPath(this.data , path)[0]);  
```

Comment:
refreshFileMetadata doesn't actually use much of the node object at all, it just uses node.path

meaning you can just pass along path and then refactor refreshFileMetadata slightly to account for this edgecase i.e. 
```
refreshFileMetadata(nodeOrPath: any) {
    let path = nodeOrPath;
    if (nodeOrPath.path) { 
      path = nodeOrPath.path;
    }

```

Doing it this way, avoids the need for findNodeByPath, which searches through all nodes within a node object to find the path match, a potentially expensive process

Reply:
Actually, this isn't true. We don't need the node object to get any of the info we need, but do we need the object to actually update the info once we got it
i.e. node.directory = 

Nevermind